public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -364,36 +443,103 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
+                        conceptAnalysis.setGenderAnalysis(TO_CLIENT_ANALYSIS.apply(aa));
+                    }else if(aa.getAnalysisId() == AGE_ANALYSIS_ID){
+                        for(AchillesResult ar: aa.getResults()){
+                            String analysisStratumName=ar.getAnalysisStratumName();
+                            if (analysisStratumName == null || analysisStratumName.equals("")) {
+                                ar.setAnalysisStratumName(QuestionConcept.ageStratumNameMap.get(ar.getStratum2()));
+                            }
+                        }
+                        conceptAnalysis.setAgeAnalysis(TO_CLIENT_ANALYSIS.apply(aa));
+                    }else if(aa.getAnalysisId() == RACE_ANALYSIS_ID){
+                        for(AchillesResult ar: aa.getResults()){
+                            String analysisStratumName=ar.getAnalysisStratumName();
+                            if (analysisStratumName == null || analysisStratumName.equals("")) {
+                                ar.setAnalysisStratumName(QuestionConcept.raceStratumNameMap.get(ar.getStratum2()));
+                            }
+                        }
+                        conceptAnalysis.setRaceAnalysis(TO_CLIENT_ANALYSIS.apply(aa));
+                    }else if(aa.getAnalysisId() == ETHNICITY_ANALYSIS_ID){
+                        for(AchillesResult ar: aa.getResults()){
+                            String analysisStratumName=ar.getAnalysisStratumName();
+                            if (analysisStratumName == null || analysisStratumName.equals("")) {
+                                ar.setAnalysisStratumName(QuestionConcept.ethnicityStratumNameMap.get(ar.getStratum2()));
+                            }
+                        }
+                        conceptAnalysis.setEthnicityAnalysis(TO_CLIENT_ANALYSIS.apply(aa));
+                    }else if(aa.getAnalysisId() == MEASUREMENT_GENDER_ANALYSIS_ID){
+
+                        List<AchillesResult> maleResults = new ArrayList<>();
+                        List<AchillesResult> femaleResults = new ArrayList<>();
+                        List<AchillesResult> otherResults = new ArrayList<>();
+
+                        for(AchillesResult ar: aa.getResults()){
+                            String analysisStratumName=ar.getAnalysisStratumName();
+                            String stratum2 = ar.getStratum2();
+                            if (analysisStratumName == null || analysisStratumName.equals("")) {
+                                ar.setAnalysisStratumName(QuestionConcept.genderStratumNameMap.get(ar.getStratum2()));
+                                if(stratum2.equals(MALE_CONCEPT_ID)){
+                                    maleResults.add(ar);
+                                }else if(stratum2.equals(FEMALE_CONCEPT_ID)){
+                                    femaleResults.add(ar);
+                                }else{
+                                    otherResults.add(ar);
```

Comment:
The other results is ok for now.  Just thinking outloud here I'll have to see how it graphs ... Basically , if we had 4 other genders , we would have 40 results here for binned data, 10 for each gender, with 4 copies of each bin with different counts.  That may not graph right. May have to loop through the other results and sum up the counts for each bin so we only have one result per bin. Know what i mean ? Something like : 

unique_other_results = []
for (r of other_results) { 
    bin = r.stratum_4;
    If bin is in unique_other_results then next; // we already summed this one
    bin_count = 0;
    for (r2 of other_results) {
          if (bin === r2.stratum_4) { bin_count += r2.count_value }
    }
    Add new result for this bin to unique_other_results

I may just do something like this on front end.  Overall we may just scrap all this male / female / other and make an array for the genders and just put each gender in it's own in the array. Eventually I think we are going to show all genders ...